src/securejoin.rs

```diff
@@ -351,6 +351,24 @@ async fn securejoin(context: &Context, qr: &str) -> ChatId {
         while !context.shall_stop_ongoing().await {
             async_std::task::sleep(Duration::from_millis(50)).await;
         }
+
+        // handle_securejoin_handshake() calls Context::stop_ongoing before the group chat
+        // is created (it is created after handle_securejoin_handshake() returns by
+        // dc_receive_imf()).  As a hack we just wait a bit for it to appear.
+        let start = Instant::now();
+        while start.elapsed() < Duration::from_secs(7) {
+            let bob = context.bob.read().await;
+            match chat::get_chat_id_by_grpid(
+                context,
+                bob.qr_scan.as_ref().unwrap().text2.as_ref().unwrap(),
+            )
+            .await
+            {
+                Ok(_) => break,
```

Comment:
Could use `.is_ok()` instead of pattern matching.

Reply:
done, also no longer holds the read lock while sleeping which is probably a bit nicer.